ethclient/ethclient.go

```diff
@@ -325,13 +325,20 @@ func (ec *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header)
 	return ec.c.EthSubscribe(ctx, ch, "newHeads")
 }
 
-// GetBlockResultByHash returns the blockResult.
+// GetBlockResultByHash returns the blockResult by hash.
 func (ec *Client) GetBlockResultByHash(ctx context.Context, blockHash common.Hash) (*types.BlockResult, error) {
-	var blockResult types.BlockResult
-	if err := ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByHash", blockHash); err != nil {
-		return nil, err
+	blockResult := &types.BlockResult{}
+	err := ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByHash", blockHash)
+	if err != nil {
+		err = ec.c.CallContext(ctx, &blockResult, "eth_getBlockResultByNumberOrHash", blockHash)
```

Comment:
shouldn't we also add the result to cache after we `eth_getBlockResultByNumberOrHash`?

Reply:
if we add the result to cache then we can combine `(ec *Client) GetBlockResultByHash` & `(ec *Client) GetBlockResultByNumber`?

Reply:
> shouldn't we also add the result to cache after we `eth_getBlockResultByNumberOrHash`?

Great idea, i will add cache in the api struct.